### DISCUSSION TITLE
VCS branch and tag schema

### DISCUSSION THREAD

**Discussion by psiryan** [apache/netbeans#3817] [Ideas]
The question was pose recently in a [recent ticket](https://github.com/apache/netbeans/issues/3779#issuecomment-1068316004) regarding the format of numbered tags and branches. I would like to begin a topic of discussion where a formal and official schema for labeling branches and tags can be established and followed. The reason is that many down stream systems may use the branch or tag sequence for automated build systems. Having a consistent and predictable pattern to follow will help maintain the integrity of those systems as changes are made upstream.

Presently it appears that branches for releases seem to follow a consistent pattern of the phrase `release` followed by the version major and minor numbers with the decimal removed: `release00.0`.

Similarly, tags have followed a pattern consisting of a pattern which contains the version major and minor, separated by a decimal, optionally followed by a hyphen and release label, e.g. `00.0-alpha`, until version 13 was released.

Version 13 is the first version to break the consistent pattern used so far in the tag schema, truncating the minor version `0` from any of the tags. Previous releases included the `0` or other minor version number in the tag name.

The proposal for discussion here is whether the schema should change to drop the minor version from tags if they are a `0` going forward.

**Comment by mbien** [comment]
NetBeans doesn't have minor versions anymore. Next version will be 14. @neilcsmith-net tagged the current release with 13. Sounds like a reasonable choice :)

For other formats, like version x10, where 12.6 would be represented by 126 for example, it will just continue: 130, 140 etc. Removing the 0 would cause conflicts in a few decades and make many devs angry (probably).

**Comment by vieiro** [comment]
Yep, this was already discussed (and approved) a long time ago in the dev mailing list, were it belongs.

Having said that, we have a  [netbeansrelease.json](https://gitbox.apache.org/repos/asf?p=netbeans-jenkins-lib.git;a=blob_plain;f=meta/netbeansrelease.json) that we use in our automated build system.

This may be also useful for those "_many down stream systems that use the branch or tag sequence for automated build systems_".
